CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+```
+##### 3. Ensure the tests pass
+```
+make test
+```
+- If your changes involve updating the expected output files run:
+```
+make test TEST_ARGS="--update-expected-output"
+```
+
+### Add a new test to CI
+  Usually, fixing an existent feature or creating a new one requires editing/adding a new test for it in CI:  `src/tests/integration/test_foundry_prove.py`
+  - If you are adding a new feature, you might want to add a new test for it in CI (see the tests for recently added features like [test_deployment_summary](https://github.com/runtimeverification/kontrol/blob/0c18ea7e846f9278624007c8072326d1ea1f95df/src/tests/integration/test_foundry_prove.py#L603) and [test_xml_report](https://github.com/runtimeverification/kontrol/blob/0c18ea7e846f9278624007c8072326d1ea1f95df/src/tests/integration/test_foundry_prove.py#L743))
+  - If you are addressing an issue that is reproducible with a Foundry test, then:
+    - Add the test into `src/tests/integration/test-data/foundry/test`
+    - Edit the `foundry-{prove-all,skip,fail}` lists according to your fix. For instance:
+      - If the test was failing with `kontrol prove` and it is supposed to pass, then add the test to `foundry-prove-all` and make sure that the test is not in `foundry-prove-skip` (or `foundry-prove-skip-legacy` in case the test should also run in the legacy backend)
+      - If the test was passing with `kontrol prove` and it is supposed to fail, then add the test to `foundry-fail` and make sure that the test is not in `foundry-prove-skip` (or `foundry-prove-skip-legacy` in case the test should also run in the legacy backend)
+  - Run `make test-integration` locally to ensure all tests pass in CI.
+  - Note: since `make test-integration` is time-expensive, before you can run your test to ensure it passes with: 
+```
+poetry run pytest src/tests/integration -k 'test_foundry_xml_report' --maxfail=1 --verbose --durations=0 --numprocesses=4 --dist=worksteal
+```
+
+### Run kontrol with a custom `pyk` / `kevm`
+- If your fix involves changes in `pyk` or `kevm`, you might want to run Kontrol with a custom `pyk`/`kevm`. If that is the case, follow these [instructions](https://github.com/runtimeverification/kontrol/issues/319).
+
+### Adding support for a new cheatcode
+The foundry success predicate as well as the addresses needed to define cheatcodes are in [foundry.md](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/foundry.md).  The rules to support cheatcodes are located in [cheatcodes.md](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/cheatcodes.md). At the top of the file there is the subconfiguration needed to implement the cheatcodes. The [structure of execution](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/cheatcodes.md#structure-of-execution) presents how calls to the Foundry cheatcode address are handled. After understanding this infrastructure, implementing a new cheatcode resumes to:
```

Comment:
I think it'd help to have a brief description of what `foundry_success` predicate is if it's referenced:
```suggestion
The `foundry_success` predicate as well as the addresses needed to define cheatcodes are in [foundry.md](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/foundry.md). The rules to support cheatcodes are located in [cheatcodes.md](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/cheatcodes.md). At the top of the file there is the subconfiguration needed to implement the cheatcodes. The [structure of execution](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/cheatcodes.md#structure-of-execution) presents how calls to the Foundry cheatcode address are handled. After understanding this infrastructure, implementing a new cheatcode resumes to:
```

Reply:
But if you open the link, you will find a description of the `foundry_success` predicate. Is it worth adding the description anyway?